WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -69,6 +69,11 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
         private bool moved;
 
         /// <summary>
+        /// The ODataResponseMessage to dispose when we are done.
+        /// </summary>
+        private IODataResponseMessage responseMessage;
```

Comment:
The message is disposed from SaveResult.CreateNextChange in the finally block. This I have validated with code like...

            var file = new File
            {
                Title = "5.8.0",
                FileClass_Value = "UT"
            };
            DataServiceCollection<File> files = new DataServiceCollection<File>(ctx);
            files.Add(file);

            ctx.SaveChanges();

Is it another scenario you have?